src/game.py

```diff
@@ -78,27 +93,119 @@ def duel(
                 + f"({chosen_spell['Description']})!"
             )
             print(
-                f"{character['Name']} has a total power rating of {battle_stats[i]}!\n"
+                f"{character['Name']} has a total power"
+                f" rating of {battle_stats[i]}!\n"
             )
 
-        # finding the winner
-
-        winner_key = max(battle_stats, key=battle_stats.get)
-
-        print(
-            f"{character_group[winner_key]['Name']} defeated "
-            + f"{character_group[abs(winner_key-1)]['Name']}!\n\n"
-        )
+        # determining the winner
+        # added scenario when the output of battle is a tie
+        if battle_stats[0] > battle_stats[1]:
+            winner_key = 0
+            print(
+                f"{character_group[winner_key]['Name']} defeated "
+                + f"{character_group[abs(winner_key-1)]['Name']}!\n\n"
+            )
+        elif battle_stats[0] == battle_stats[1]:
+            winner_key = 0
+            print(
+                f"{character_group[winner_key]['Name']} cast the spell "
```

Comment:
This is a cool way to handle ties

Reply:
I implemented that in one of my drafts. I just wasn't sure to include it. I'm glad Wafi did.